src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -801,17 +803,38 @@ public override int GetHashCode()
 
         public bool Equals(Guid g) => EqualsCore(this, g);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool EqualsCore(in Guid left, in Guid right)
         {
-            ref int rA = ref Unsafe.AsRef(in left._a);
-            ref int rB = ref Unsafe.AsRef(in right._a);
+            if (Sse2.IsSupported)
```

Comment:
What about ARM64? We care about ARM64 as much as we care about x64.

Reply:
Well, if `SSE2` is not supported, we can just use a method without `SSE2`?